src/app_unit.rs

```diff
@@ -244,9 +242,9 @@ impl Au {
     #[inline]
     pub fn from_f32_px(px: f32) -> Au {
         let float = (px * AU_PER_PX as f32).round();
-        if float > MAX_AU.0 as f32 {
+        if (float as f64) > MAX_AU.0 as f64 {
```

Comment:
Rather than casting to `f64` twice in conditions, probably making `float` an `f64` is easier and more precise?